### ISSUE TITLE
Confusing JPEG API documentation

### ISSUE THREAD

**Issue by Shnatsel** [etemesi254/zune-image#100]
It's not clear if [`get_output_colorspace`](https://docs.rs/zune-jpeg/0.3.12/zune_jpeg/struct.JpegDecoder.html#method.get_output_colorspace) can be called only after `decode` or if `decode_headers` is sufficient.

[`info`](https://docs.rs/zune-jpeg/0.3.12/zune_jpeg/struct.JpegDecoder.html#method.info) refers to `decode_file` or `decode_buffer` functions which do not exist.

Also, this kind of requirements can be encoded in the type system using a pattern called [typestate](http://cliffle.com/blog/rust-typestate/). But that's out of scope of this issue, I just want clarity on the documentation.

**Comment by etemesi254** [comment]
Will fix it later today

**Comment by etemesi254** [comment]
Fixed, waiting for a release, being blocked by me trying to figure out how to do https://github.com/etemesi254/zune-image/issues/101

**Comment by etemesi254** [comment]
> It's not clear if [get_output_colorspace](https://docs.rs/zune-jpeg/0.3.12/zune_jpeg/struct.JpegDecoder.html#method.get_output_colorspace) can be called only after decode or if decode_headers is sufficient.

Added in 0.3.13

Calling `decode_headers` is sufficient to know what colorspace the output is, if this is called after `decode` it indicates the colorspace the output is currently in.

> [info](https://docs.rs/zune-jpeg/0.3.12/zune_jpeg/struct.JpegDecoder.html#method.info) refers to decode_file or decode_buffer functions which do not exist.

Removed

> Also, this kind of requirements can be encoded in the type system using a pattern called [typestate](http://cliffle.com/blog/rust-typestate/). But that's out of scope of this issue, I just want clarity on the documentation.

Would be interesting to see a project that implements it
